js/src/client.ts

```diff
@@ -1625,6 +1627,33 @@ export class Client implements LangSmithTracingClientInterface {
           ...this.fetchOptions,
         }
       );
+    };
+
+    try {
+      let res: Response;
+
+      // attempt stream only if not disabled and not using node-fetch
+      if (!isNodeFetch && !this.multipartStreamingDisabled) {
```

Comment:
As discussed offline, to be completely safe and avoid race conditions let's keep the local state of whether the request was already tried with streaming